build.sbt

```diff
@@ -1,5 +1,8 @@
 name := """dhall"""
 version := "1.0"
 scalaVersion := "2.11.7"
-libraryDependencies += "org.specs2" %% "specs2-core" % "3.8.9" % "test"
-
+libraryDependencies ++= Seq(
+  "org.specs2" %% "specs2-core" % "3.8.9" % "test",
+  "org.typelevel" %% "cats" % "0.9.0",
+  compilerPlugin("org.spire-math" %% "kind-projector" % "0.9.3")
```

Comment:
I suggest removing `kind-projector`. It is not native Scala syntax, and it throws off IDEA (and I suppose all other major IDEs). 

It is quite easily to replace usage of this plugin using type-level functions. See the following example.

```scala

// with kind projector 

new Monad[Expr[S, ?]] { /**/ }

// without kind projector

trait Partial2[F[_, _], A] {     // This can go in utils
  type Apply[B] = F[A, B]
}

new Monad[Partial2[Expr, S]#Apply] { /**/ }
```